controllers/openstackmachine_controller.go

```diff
@@ -343,19 +378,23 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 	switch instanceStatus.State() {
 	case infrav1.InstanceStateActive:
 		scope.Logger.Info("Machine instance is ACTIVE", "instance-id", instanceStatus.ID())
+		conditions.MarkTrue(openStackMachine, infrav1.InstanceReadyCondition)
 		openStackMachine.Status.Ready = true
 	case infrav1.InstanceStateError:
 		// Error is unexpected, thus we report error and never retry
 		handleUpdateMachineError(scope.Logger, openStackMachine, errors.Errorf("OpenStack instance state %q is unexpected", instanceStatus.State()))
+		conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.InstanceStateErrorReason, clusterv1.ConditionSeverityError, "")
 		return ctrl.Result{}, nil
 	case infrav1.InstanceStateDeleted:
 		// we should avoid further actions for DELETED VM
 		scope.Logger.Info("Instance state is DELETED, no actions")
+		conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.InstanceDeletedReason, clusterv1.ConditionSeverityError, "")
```

Comment:
This might be an expected state, so don't know if we should really considers this error severity.